doc/source/c-api.rst

```diff
@@ -25,6 +25,8 @@ Types
     :project: TileDB-C
 .. doxygentypedef:: tiledb_query_t
     :project: TileDB-C
+.. doxygentypedef:: tiledb_subarray_t
```

Comment:
The `*_2()` functions that take subarrays aren't documented in this file, but maybe that's ok if the API is in flux.